### PR TITLE
Update README for Metrics Collector ver. 2.0

### LINKED ISSUE #56: Update README for MC 2.0
- Update all related links and references to SCSt 2.x and Spring Boot 2.x
- Explain the switch to Micrometer
- Explain the support for SCSt 1.x AND SCSt 2.x metrics

### PR DESCRIPTION
**Pull request by tzolov**
Resolves #56

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -6,26 +6,29 @@ The metrics collector is a companion application to the http://cloud.spring.io/s
 It collects metrics emitted by http://cloud.spring.io/spring-cloud-stream/[Spring Cloud Stream] apps, and groups them together around the stream definition that dataflow used to deploy them.
 
 All of the OOB http://cloud.spring.io/spring-cloud-stream-app-starters/[Spring Cloud App Starters] currently have the metrics emitter module already bundled on it.
-You can enable metric emission by setting a flag explained http://docs.spring.io/spring-cloud-stream/docs/Chelsea.SR1/reference/htmlsingle/index.html#_metrics_emitter[here]
+You can enable metric emission by setting the destination name for metrics binding, e.g. `spring.cloud.stream.bindings.applicationMetrics.destination=<DESTINATION_NAME>`
+
+Starting with Spring Cloud Stream 2.0 the metrics has been switched to use https://micrometer.io/[Micrometer]. See https://docs.spring.io/spring-cloud-stream/docs/Elmhurst.RELEASE/reference/htmlsingle/#spring-cloud-stream-overview-metrics-emitter [Metrics Emitter] for more details
```
Comment (sabbyanandan) [inline README.adoc]: How about:

> Starting with Spring Cloud Stream 2.0, the default metrics support has been switched to use https://micrometer.io/[Micrometer]. See https://docs.spring.io/spring-cloud-stream/docs/Elmhurst.RELEASE/reference/htmlsingle/#spring-cloud-stream-overview-metrics-emitter [Metrics Emitter] for more details.
+For information on how to configure the metrics-emitter for Spring Cloud Stream 1.x applications, see https://docs.spring.io/spring-cloud-stream/docs/Ditmars.SR3/reference/htmlsingle/#_metrics_emitter[here].

File: README.adoc [hunk 2]
```diff
@@ -6,26 +6,29 @@ The metrics collector is a companion application to the http://cloud.spring.io/s
...
+Information how to configure the metrics for legacy Spring Cloud Stream 1.x applications is provided https://docs.spring.io/spring-cloud-stream/docs/Ditmars.SR3/reference/htmlsingle/#_metrics_emitter[here].
+
+The metrics collector 2.x can collect and process metrics coming from Spring Cloud Stream 1.x and 2.x applications.
```
Comment (sabbyanandan) [inline README.adoc]: "from both Spring Cloud Stream 1.x and 2.x applications. "

File: README.adoc [hunk 3]
```diff
@@ -6,26 +6,29 @@ The metrics collector is a companion application to the http://cloud.spring.io/s
...
-|metrics-collector-rabbit| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-rabbit/1.0.0.RELEASE/metrics-collector-rabbit-1.0.0.RELEASE.jar| https://hub.docker.com/r/springcloud/metrics-collector-rabbit/tags/ [docker pull metrics-collector-rabbit:1.0.0.RELEASE]
-|metrics-collector-kafka-09| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-kafka-09/1.0.0.RELEASE/metrics-collector-kafka-09-1.0.0.RELEASE.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka-09/tags/ [docker pull metrics-collector-kafka-09:1.0.0.RELEASE]
-|metrics-collector-kafka-10| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-kafka-10/1.0.0.RELEASE/metrics-collector-kafka-10-1.0.0.RELEASE.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka-10/tags/ [docker pull metrics-collector-kafka-10:1.0.0.RELEASE]
+|metrics-collector-rabbit| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-rabbit/2.0.0.RELEASE/metrics-collector-rabbit-2.0.0.RELEASE.jar| https://hub.docker.com/r/springcloud/metrics-collector-rabbit/tags/ [docker pull metrics-collector-rabbit:2.0.0.RELEASE]
```
Comment (sabbyanandan) [inline README.adoc]: We can't add `2.0.0.RELEASE` tags yet. We will be at 2.0.0.M1 at the beginning.

File: README.adoc [hunk 4]
```diff
@@ -6,26 +6,29 @@ The metrics collector is a companion application to the http://cloud.spring.io/s
...
-|metrics-collector-kafka-09| http://repo.spring.io/libs-snapshot-local/org/springframework/cloud/metrics-collector-kafka-09/1.0.1.BUILD-SNAPSHOT/metrics-collector-kafka-09-1.0.1.BUILD-SNAPSHOT.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka-09/tags/ [docker pull metrics-collector-kafka-09:1.0.1.BUILD-SNAPSHOT]
-|metrics-collector-kafka-10| http://repo.spring.io/libs-snapshot-local/org/springframework/cloud/metrics-collector-kafka-10/1.0.1.BUILD-SNAPSHOT/metrics-collector-kafka-10-1.0.1.BUILD-SNAPSHOT.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka-10/tags/ [docker pull metrics-collector-kafka-10:1.0.1.BUILD-SNAPSHOT]
+|metrics-collector-rabbit| http://repo.spring.io/libs-snapshot-local/org/springframework/cloud/metrics-collector-rabbit/2.0.1.BUILD-SNAPSHOT/metrics-collector-rabbit-2.0.1.BUILD-SNAPSHOT.jar| https://hub.docker.com/r/springcloud/metrics-collector-rabbit/tags/ [docker pull metrics-collector-rabbit:2.0.1.BUILD-SNAPSHOT]
```
Comment (sabbyanandan) [inline README.adoc]: Same. We haven't released 2.0 yet, and `2.0.1.BUILD-SNAPSHOT` is a stretch.

File: README.adoc [hunk 5]
```diff
@@ -79,29 +81,30 @@ Assuming apps have been deployed configuring `spring.cloud.stream.bindings.appli
 
 === Controlling eviction
 
-Internally the collector maintains a cache of the metrics it receives. The default for metric emission is every 5 seconds, but can be tuned on the application by using Spring Boot's metrics exporter scheduling control, please refer to the docs http://docs.spring.io/spring-cloud-stream/docs/Chelsea.SR1/reference/htmlsingle/index.html#_metrics_emitter[here] to configure your applications.
+Internally the collector maintains a cache of the metrics it receives. The default for metric emission is every 60 seconds (for Spring Boot 1.x applications it is every 5 seconds), but can be tuned on the application by using Spring Boot's metrics exporter scheduling control, please refer to the docs https://docs.spring.io/spring-cloud-stream/docs/Elmhurst.RELEASE/reference/htmlsingle/index.html#spring-cloud-stream-overview-metrics-emitter[here] to configure your applications.
```
Comment (sabbyanandan) [inline README.adoc]: Does it mean, we will see throughput rate changes only after every 60s in SCDF UI? Or, is it about the cache TTL?

File: README.adoc [hunk 6]
```diff
@@ -79,29 +81,30 @@ Assuming apps have been deployed configuring `spring.cloud.stream.bindings.appli
...
 
-The default for the collector is to evict any metric reading that was not updated over the past 30 seconds.
+The default for the collector is to evict any metric reading that was not updated over the past 90 seconds.
```
Comment (sabbyanandan) [inline README.adoc]: How will this change the UI behavior? We need to update it in the SCDF docs.

File: README.adoc (hunk 1 above)
Comment (tzolov) [reply README.adoc]: +1

File: README.adoc (hunk 3 above)
Comment (tzolov) [reply README.adoc]: ah, i assumed we are making a GA release. Will fix the versions

File: README.adoc (hunk 5 above)
Comment (tzolov) [reply README.adoc]: Yes!
If the metrics emitted from the SCSt apps come every 60 sec. (this is the default so SCSt2.x) then the UI will be able to show changes at 60 sec rate. 
The important bit is to align the MC's cache retention interval to be bigger then the default metric emission interval. That is the reason we have increased it from 30 to 90 seconds.  If the MC's cache retention time is smaller then the emission time then the UI will experience disappearance and reappearance of the metrics visualization.

Note that all the time intervals can be changed at deployement. 
- for emission time interval: `spring.cloud.stream.metrics. scheduleInterval`
- for MC retention time: `spring.cloud.dataflow.metrics.collector.evictionTimeout`

File: README.adoc (hunk 6 above)
Comment (tzolov) [reply README.adoc]: As explained above if the eviction timeout is lower that the emission interval the UI metrics visualization will became unstable (comes an goes...)

File: README.adoc [hunk 7]
```diff
@@ -63,45 +65,45 @@ You should find the *uber jar* of the collector inside your `target` folder.
...
 Server:
-java -jar spring-cloud-dataflow-server-local/target/spring-cloud-dataflow-server-local-1.2.0.BUILD-SNAPSHOT.jar --spring.cloud.dataflow.metrics.collector.uri=http://localhost:8080 --spring.cloud.dataflow.metrics.collector.username=spring --spring.cloud.dataflow.metrics.collector.password=cloud
+java -jar spring-cloud-dataflow-server-local/target/spring-cloud-dataflow-server-local-1.5.0.BUILD-SNAPSHOT.jar --spring.cloud.dataflow.metrics.collector.uri=http://localhost:8080 --spring.cloud.dataflow.metrics.collector.username=spring --spring.cloud.dataflow.metrics.collector.password=cloud
 
-Register Apps:
-app import --uri http://bit.ly/Bacon-RELEASE-stream-applications-rabbit-maven
+Register SCSt 1.x Apps:
+app import --uri http://bit.ly/Celsius-SR1-stream-applications-rabbit-maven
 
+Or Register SCSt 2.x Apps: TODO
```
Comment (sabbyanandan) [inline README.adoc:106]: We can remove it for now. Once the App release is done, we can add it with links.

**Review by sabbyanandan — APPROVED** [review]
Latest changes look good except the 2.0 M1 links, which of course will only be available after the release. I think that's OK. :)

File: README.adoc (hunk 2 above)
Comment (markpollack) [reply README.adoc]: saying 'legacy' feels over the top....

File: README.adoc (hunk 3 above)
Comment (markpollack) [reply README.adoc]: also it will be libs-milestone not libs-release

File: README.adoc [hunk 8]
```diff
@@ -6,26 +6,29 @@ The metrics collector is a companion application to the http://cloud.spring.io/s
...
-|metrics-collector-kafka-10| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-kafka-10/1.0.0.RELEASE/metrics-collector-kafka-10-1.0.0.RELEASE.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka-10/tags/ [docker pull metrics-collector-kafka-10:1.0.0.RELEASE]
+|metrics-collector-rabbit| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-rabbit/2.0.0.RELEASE/metrics-collector-rabbit-2.0.0.RELEASE.jar| https://hub.docker.com/r/springcloud/metrics-collector-rabbit/tags/ [docker pull metrics-collector-rabbit:2.0.0.RELEASE]
+|metrics-collector-kafka| http://repo.spring.io/libs-release/org/springframework/cloud/metrics-collector-kafka/2.0.0.RELEASE/metrics-collector-kafka-2.0.0.RELEASE.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka/tags/ [docker pull metrics-collector-kafka:2.0.0.RELEASE]
 |======================
```
Comment (markpollack) [inline README.adoc]: same thing - libs-milestone and 2.0 M1

File: README.adoc (hunk 4 above)
Comment (markpollack) [reply README.adoc]: Can reference 2.0.0.BUILD-SNAPSHOT

File: README.adoc [hunk 9]
```diff
@@ -6,26 +6,29 @@ The metrics collector is a companion application to the http://cloud.spring.io/s
...
-|metrics-collector-kafka-10| http://repo.spring.io/libs-snapshot-local/org/springframework/cloud/metrics-collector-kafka-10/1.0.1.BUILD-SNAPSHOT/metrics-collector-kafka-10-1.0.1.BUILD-SNAPSHOT.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka-10/tags/ [docker pull metrics-collector-kafka-10:1.0.1.BUILD-SNAPSHOT]
+|metrics-collector-rabbit| http://repo.spring.io/libs-snapshot-local/org/springframework/cloud/metrics-collector-rabbit/2.0.1.BUILD-SNAPSHOT/metrics-collector-rabbit-2.0.1.BUILD-SNAPSHOT.jar| https://hub.docker.com/r/springcloud/metrics-collector-rabbit/tags/ [docker pull metrics-collector-rabbit:2.0.1.BUILD-SNAPSHOT]
+|metrics-collector-kafka| http://repo.spring.io/libs-snapshot-local/org/springframework/cloud/metrics-collector-kafka/2.0.1.BUILD-SNAPSHOT/metrics-collector-kafka-2.0.1.BUILD-SNAPSHOT.jar | https://hub.docker.com/r/springcloud/metrics-collector-kafka/tags/ [docker pull metrics-collector-kafka:2.0.1.BUILD-SNAPSHOT]
```
Comment (markpollack) [inline README.adoc]: reference 2.0.0.BUILD-SNAPSHOT

File: README.adoc (hunk 6 above)
Comment (markpollack) [reply README.adoc]: No impact on UI behavior using the out of the box settings.

**Comment by markpollack** [comment]
squashed and did some editing.  Made download sections for both 1.0 GA and 2.0 M1

merged as 45e3120b502c71b85ac9bbc10cd77125bc7431f8

**Pull request closed without merging**